website/docs/guides/customize-schema-alias.md

```diff
@@ -0,0 +1,490 @@
+
+:::
+Out-of-the-box, the default behavior of dbt is the following:
+
+- The database where the object is created will be the database configured at the [environment level in dbt Cloud](/docs/dbt-cloud-environments), or in the `profiles.yml` in dbt Core.
+
+- The schema depends on whether a [custom schema](/docs/build/custom-schemas) has been defined for the model:
+    - If no custom schema is defined, the object will be created in the schema configured in dbt Cloud (`dbt_username` for dev and the default schema for deployment environments) or the schema in the `profiles.yml` in dbt Core.
+ 
+
+- Note that for automated CI jobs, the schema name is derived from the job number and PR number: `dbt_cloud_pr_<job_id>_<pr_id>`
+- If a custom schema has been defined, it will concatenate the schema from above with the custom one.
+- For example, if the configured schema is `dbt_myschema` and the custom one is `marketing`, the objects will be created under `dbt_myschema_marketing`.
+
+- The object name depends on whether an [alias](/reference/resource-configs/alias) has been defined on the model:
+    - If no alias is defined, the object will be created with the same name as the model, without the `.sql` or `.py` at the end.
+    - If an alias is defined, the object will be created with the configured alias.
+
+Those default rules are a great starting point, and many organizations choose to stick with those without any customization required.
+
+The defaults allow developers to work in their own isolated schemas (sandboxes) without overwriting each other's work &mdash; even if they're working on the same tables.
+
+
+## How to customize this behavior
+
+While the default behavior will fit the needs of most organizations, there are occasions where this approach won't jive.
+
+For example, dbt expects that it has permission to create schemas on the fly (and we recommend that the users running dbt have this ability), but it might not be allowed at your company.
+
+Or, based on how you've designed your warehouse, you may wish to minimize the number of schemas in your dev environment (and avoid schema sprawl by not creating the combination of all developer schemas and custom schemas).
+
+Alternatively, you may even want your dev schemas to be named after feature branches instead of the developer name.
+
+For this reason, dbt offers three macros to customize what objects are created in the Data Warehouse:
+
+- [`generate_database_name()`](/docs/build/custom-databases#generate_database_name)
+- [`generate_schema_name()`](/docs/build/custom-schemas#how-does-dbt-generate-a-models-schema-name)
+- [`generate_alias_name()`](/docs/build/custom-aliases#generate_alias_name)
+
+By overwriting one or multiple of those macros, we can tailor where dbt objects are created in the data warehouse and align with any existing requirement.
+
+:::note Key concept
```

Comment:
love this key concept 🔥! great example! i wonder if we should add line breaks in the callout so it's easier to read?